lib/cfn-model/util/wildcard_patterns.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+# Create array of wildcard patterns for a given input string
+
+def wildcard_patterns(input, pattern_types: %w[front back both])
+  input_string = input.to_s
+  results = [input_string]
+  pattern_types.each do |pattern_type|
+    case pattern_type
+    when 'front'
+      results += wildcard_front(input_string)
+    when 'back'
+      results += wildcard_back(input_string)
+    when 'both'
+      results += wildcard_front_back(input_string)
+    else
+      raise "no pattern of type: #{pattern_type}. Use one or more of: front, back, both"
+    end
+  end
+  results + ['*']
+end
+
+private
+
+def wildcard_back(input_string, results = [], prepend: '')
```

Comment:
i'd avoid doing an assignment in the argument list.  rubocop might complain too?

Reply:
I believe that is just a way of defaulting the value of `results` to an empty array.

rubocop doesn't seem to mind.
```
$ rubocop lib/cfn-model/util/wildcard_patterns.rb
.rubocop.yml: Metrics/LineLength has the wrong namespace - should be Layout
Inspecting 1 file
C

Offenses:

lib/cfn-model/util/wildcard_patterns.rb:5:1: C: Metrics/MethodLength: Method has too many lines. [15/10]
def wildcard_patterns(input, pattern_types: %w[front back both]) ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
$
```

Reply:
ok - the default argument stuff is ok - but for my own double vision - let's not mix named arguments with defaults with positional arguments with default.

Reply:
Sounds good.  Updated in latest commit.